### PR TITLE
[JFT-05] - Fix routing across the app

### PR DESCRIPTION
**Pull request by julianfigueroaGL**
This ticket solves problems with routing in the Dashboard page and for the Detail page

JFT-05 Fix Routing

JFT-05 Menu Drawe Collapsible

### CODE REVIEW

File: src/components/recent-podcast/script.js [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
 import FavoriteButton from '@/components/favorite-button/favorite-button.vue';
-import { DETAIL_PATH } from '@/router/index';
 
 export default {
   name: 'recent-podcast',
   data() {
-    return {
-      detailPath: DETAIL_PATH,
-    };
+    return {};
```
Comment (alejandrobenjumeaGL) [inline src/components/recent-podcast/script.js]: Please remove empty return. I suggest removing data() prop

File: src/router/index.js [hunk 2]
```diff
@@ -12,9 +12,9 @@ const routes = [
     component: Home,
   },
   {
-    path: '/detail/:id',
-    name: 'Detail',
-    component: () => import(/* webpackChunkName: "detail" */ '../views/details/details.vue'),
+    path: '/detail/:id?',
+    name: 'detail',
```
Comment (alejandrobenjumeaGL) [inline src/router/index.js]: please use detail or details. I recommend calling it details

File: src/views/home/home.js [hunk 3]
```diff
@@ -27,7 +27,15 @@ export default {
   },
   computed: {
     toList() {
-      return `/list/${this.category}`;
+      return {
+        name: 'List',
```
Comment (alejandrobenjumeaGL) [inline src/views/home/home.js]: for path-naming use lowercase

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]
Please remove **DETAIL_PATH** from `@/router/index` file
Also, for navigation I recommend having a back button inside other pages

File: src/router/index.js (hunk 2 above)
Comment (julianfigueroaGL) [reply src/router/index.js]: done

File: src/views/home/home.js (hunk 3 above)
Comment (julianfigueroaGL) [reply src/views/home/home.js]: done

File: src/components/recent-podcast/script.js (hunk 1 above)
Comment (julianfigueroaGL) [reply src/components/recent-podcast/script.js]: done

Comment (julianfigueroaGL) [reply src/components/recent-podcast/script.js]: done

File: src/router/index.js (hunk 2 above)
Comment (julianfigueroaGL) [reply src/router/index.js]: done

File: src/views/home/home.js (hunk 3 above)
Comment (julianfigueroaGL) [reply src/views/home/home.js]: done

File: src/components/list-podcast/script.js [hunk 4]
```diff
@@ -50,10 +50,7 @@ export default {
       default: '100%',
     },
     to: {
-      type: String,
-    },
-    detailPath: {
-      type: String,
+      type: Object,
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/script.js:53]: 1. **to** property could have both types, [String, Object]
2. Are we using  **to** property somewhere?

File: src/router/index.js [hunk 5]
```diff
@@ -8,27 +8,27 @@ Vue.use(VueRouter);
...
     path: '/',
-    name: 'Dashboard',
+    name: 'dashboard',
     component: Home,
   },
   {
-    path: '/detail/:id',
-    name: 'Detail',
-    component: () => import(/* webpackChunkName: "detail" */ '../views/details/details.vue'),
+    path: '/details/:id?',
```
Comment (alejandrobenjumeaGL) [inline src/router/index.js]: is **id** really optional?

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/components/menu-drawer/menu-drawer.vue [hunk 6]
```diff
@@ -1,5 +1,18 @@
 <template>
-  <v-navigation-drawer v-model="drawer" app clipped>
+  <v-navigation-drawer v-model="drawer" app clipped :mini-variant.sync="mini"
```
Comment (alejandrobenjumeaGL) [inline src/components/menu-drawer/menu-drawer.vue]: Please format this line, use one line per parameter

`
<v-navigation-drawer
  :mini-variant.sync="mini"
  app
  clipped
  permanent
  v-model="drawer"
>
`

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/views/home/home.js [hunk 7]
```diff
@@ -20,14 +19,21 @@ export default {
...
     toList() {
-      return `/list/${this.category}`;
+      return {
+        name: 'list',
+        params: this.category,
+      };
+    },
+    toFavorite() {
+      return {
+        path: 'favorites',
```
Comment (alejandrobenjumeaGL) [inline src/views/home/home.js]: please use **name**, **path** will behaves as relative routing

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/components/list-podcast/script.js (hunk 4 above)
Comment (julianfigueroaGL) [reply src/components/list-podcast/script.js:53]: 1.  **To** prop will always receive an object
2. To will always be used in the the navigation of the button More

File: src/router/index.js (hunk 5 above)
Comment (julianfigueroaGL) [reply src/router/index.js]: Fixed

File: src/components/menu-drawer/menu-drawer.vue (hunk 6 above)
Comment (julianfigueroaGL) [reply src/components/menu-drawer/menu-drawer.vue]: done

File: src/views/home/home.js (hunk 7 above)
Comment (julianfigueroaGL) [reply src/views/home/home.js]: done

**Review by alejandrobenjumeaGL — APPROVED** [review]

**Pull request merged**